### PR TITLE
Ensure good defaults on blank c/storage configuration

### LINKED ISSUE #3952: Podman run fails to run when using named volume
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
Podman runtime error when attempting to run container with named volume. Does not seem to matter if ```podman run``` is creating the volume or ```podman volume create```. Does not matter whether container is a custom build or remote image. 

**Steps to reproduce the issue:**

1. Run any image, specifying a named volume, e.g. ```podman run -it --rm -v test:/test hello-world```

2. Get error output

3. Be sad.

**Describe the results you received:**
```
root@rocinante:~ # podman run -it --rm -v test:/test hello-world
Trying to pull docker.io/library/hello-world...
Getting image source signatures
Copying blob 1b930d010525 done
Copying config fce289e99e done
Writing manifest to image destination
Storing signatures
Error: container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/containers/storage/btrfs-containers/a261adb857d24bc7326cec1fa0d67c74465e86b1b0ecb1d59a6d2a0e2c74f9f4/userdata/volumes/test/_data\\\" to rootfs \\\"/var/lib/containers/storage/btrfs/subvolumes/36a8a9d4e1c523dba11a9a89f599d06b8689ac1ab0508f48edd8e0f205251b3a\\\" at \\\"/test\\\" caused \\\"stat /var/lib/containers/storage/btrfs-containers/a261adb857d24bc7326cec1fa0d67c74465e86b1b0ecb1d59a6d2a0e2c74f9f4/userdata/volumes/test/_data: no such file or directory\\\"\"": OCI runtime error
```

**Describe the results you expected:**
Container to mount with persistent named volume attached at specified mount point, world domination ensues shortly thereafter...

**Additional information you deem important (e.g. issue happens only occasionally):**
Consistent behavior every time on this machine.

**Output of `podman version`:**

```
podman version 1.5.0
```

**Output of `podman info --debug`:**

```
root@rocinante:~ # podman info --debug 
debug:
  compiler: gc
  git commit: ""
  go version: go1.12.9
  podman version: 1.5.0
host:
  BuildahVersion: 1.10.1
  Conmon:
    package: conmon-2.0.0-2.1.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.0, commit: unknown'
  Distribution:
    distribution: '"opensuse-tumbleweed"'
    version: "20190829"
  MemFree: 3336896512
  MemTotal: 8254947328
  OCIRuntime:
    package: runc-1.0.0~rc8-1.4.x86_64
    path: /usr/sbin/runc
    version: |-
      runc version 1.0.0-rc8
      spec: 1.0.1-dev
  SwapFree: 2147483648
  SwapTotal: 2147483648
  arch: amd64
  cpus: 4
  eventlogger: file
  hostname: rocinante
  kernel: 5.2.10-1-default
  os: linux
  rootless: false
  uptime: 1h 10m 11.21s (Approximately 0.04 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 1
  GraphDriverName: btrfs
  GraphOptions: null
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Build Version: 'Btrfs v5.2.1 '
    Library Version: "102"
  ImageStore:
    number: 21
  RunRoot: /var/run/containers/storage
  VolumePath: volumes

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
root@rocinante:~ # rpm -q podman
podman-1.5.0-1.1.x86_64

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Lenovo Thinkpad Flex 15, bare metal install

Thanks in advance!

### PR DESCRIPTION
**Pull request by mheon**
If c/storage paths are explicitly set to "" (the empty string) it will use compiled-in defaults. However, it won't tell us this via `storage.GetDefaultStoreOptions()` - we just get the empty string (which can put our defaults, some of which are relative to c/storage, in a bad spot).

Hardcode a sane default for cases like this. Furthermore, add some sanity checks to paths, to ensure we don't use relative paths for core parts of libpod.

Fixes #3952

### CODE REVIEW

File: libpod/runtime.go [hunk 1]
```diff
@@ -281,10 +281,16 @@ func defaultRuntimeConfig() (RuntimeConfig, error) {
 	if err != nil {
 		return RuntimeConfig{}, err
 	}
+	graphRoot := storeOpts.GraphRoot
+	if graphRoot == "" {
+		graphRoot = "/var/lib/containers/storage"
```
Comment (rhatdan) [inline libpod/runtime.go]: You should at least warn on this.

File: libpod/runtime.go [hunk 2]
```diff
@@ -789,6 +795,20 @@ func probeConmon(conmonBinary string) error {
 // Make a new runtime based on the given configuration
 // Sets up containers/storage, state store, OCI runtime
 func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
+	// Let's sanity-check some paths first.
+	// Relative paths can cause nasty bugs, because core paths we use could
+	// shift between runs (or even parts of the program - the OCI runtime
+	// uses a different working directory than we do, for example.
+	if !filepath.IsAbs(runtime.config.StaticDir) {
```
Comment (rhatdan) [inline libpod/runtime.go]: Should these checks should go into containers/storage?

Comment (mheon) [reply libpod/runtime.go]: These paths are all ours - libpod static and temporary files directories and libpod named volumes

File: libpod/runtime.go (hunk 1 above)
Comment (mheon) [reply libpod/runtime.go]: Good point, added warning logs

Comment (TomSweeneyRedHat) [reply libpod/runtime.go]: Concur, but could be talked into a debug instead.  Chocolate chip cookies would help sway me!  ;^)

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests.

**Pull request merged**